MoreLinq/Collections/Dictionary.cs

```diff
@@ -0,0 +1,82 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2020 Atif Aziz, Leandro F. Vieira (leandromoh). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Collections
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+
+    /// <summary>
+    /// A minimal <see cref="System.Collections.Generic.Dictionary{TKey,TValue}"/> wrapper that
+    /// allows null keys when <typeparamref name="TKey"/> is a
+    /// reference type.
+    /// </summary>
+
+    // Add members if and when needed to keep coverage.
+
+    struct Dictionary<TKey, TValue>
```

Comment:
I can agree with 3, but 1 and 2 have lost their argument with when value tuples were made mutable. See bullet point 2 of the following passage from the [remarks section](https://docs.microsoft.com/en-us/dotnet/api/system.valuetuple?view=netcore-3.1#remarks) of [`ValueTuple` documentation](https://docs.microsoft.com/en-us/dotnet/api/system.valuetuple?view=netcore-3.1):

> Value tuples are tuple types introduced in the .NET Framework 4.7 to provide the runtime implementation of [tuples in C#](https://docs.microsoft.com/en-us/dotnet/csharp/tuples) and struct tuples in F#. They differ from the tuple classes, such as [`Tuple<T1>`](https://docs.microsoft.com/en-us/dotnet/api/system.tuple-1?view=netcore-3.1), [`Tuple<T1,T2>`](https://docs.microsoft.com/en-us/dotnet/api/system.tuple-2?view=netcore-3.1), etc., as follows:
>
> - They are structures (value types) rather than classes (reference types).
> - They are mutable rather than read-only. That is, the value of tuple components can change.
> - Their data members, such as `Item1`, `Item2`, etc., are fields rather than properties.

I am not saying that this suddenly justifies making all `struct` mutable. Those have been around for a long time for sake of optimisation, such as collection enumerators. Here, this dictionary is a mere wrapper gateway to the underlying implementation, allowing nulls as an extension. It is also a very incomplete implementation as per the comment:

```c#
    // Add members if and when needed to keep coverage.
```

Reply:
I feel you did not answer the question "why a struct"? My impression is that things are usually classes by default, unless one needs either value semantics or the performance gains. Neither of these reasons is present here, so I wonder why did you go with struct.

Reply:
The `struct`-based dictionary is not a new type or abstraction per se so normal considerations like classes-by-default or value semantics aren't being strictly applied. It's there solely to avoid an extra allocation so the answer is performance by way of avoiding allocation. If you think this is not a satisfactory justification than I am happy to go with a class instead.

Reply:
I think I do prefer that.

This is the only issue left. After this, LGTM

Reply:
Done with 861585cbb9de86ab4136080698e34a25a186b941. If you could give it your final blessings by approving the PR then I'll go ahead and merge.

Thanks a ton for your review and feedback!